backend/services/tasks/pokemon_tasks.py

```diff
@@ -0,0 +1,54 @@
+from celery.utils.log import get_task_logger
+
+from pokebattle.celery import app as celery_app
+from pokemons.services import get_all_pokemons, get_pokemon_by_name
+from pokemons.models import Pokemon
+
+logger = get_task_logger(__name__)
+
+
+@celery_app.task
+def save_all_pokemons_from_pokeapi():
+    pokemons = get_all_pokemons()
+    pokemon_names = set(p["name"] for p in pokemons)
+    existing_pokemon_names = set(
+        Pokemon.objects.filter(name__in=pokemon_names).values_list("name", flat=True)
+    )
+    missing_pokemon_names = pokemon_names - existing_pokemon_names
+
+    pokemons_to_be_created = []
+    for name in missing_pokemon_names:
```

Comment:
@fjsj, I'm not sure if this and the next for are right. I added them because after the requests witl all pokemons, I need to request the specific data for each pokemon, so I can use it. Could you please let me know if I can use a better approach here?

Reply:
yes, I think you need to make one request for each pokemon. But at least now you're making only 1 DB query before.